plugins/outputs/application_insights/application_insights.go

```diff
@@ -0,0 +1,410 @@
+	setFloat64TelemetryProperty(&(telemetry.Max), []string{"max"}, metric, &usedFields)
+	setFloat64TelemetryProperty(&(telemetry.StdDev), []string{"stddev"}, metric, &usedFields)
+	setFloat64TelemetryProperty(&(telemetry.Variance), []string{"variance"}, metric, &usedFields)
+
+	addRemainingMeasurements(telemetry, metric, usedFields)
+
+	return telemetry
+}
+
+func setFloat64TelemetryProperty(
+	telemetryProperty *float64,
+	candidateFields []string,
+	metric telegraf.Metric,
+	usedFields *[]string) error {
+
+	for _, fieldName := range candidateFields {
+		if metric.HasField(fieldName) {
+			metricValue, err := toFloat64(metric.Fields()[fieldName])
+			if err != nil {
+				return err
+			}
+
+			*telemetryProperty = metricValue
+
+			if usedFields != nil {
+				*usedFields = append(*usedFields, fieldName)
+			}
+
+			return nil
+		}
+	}
+
+	return fmt.Errorf("No field from the candidate list was found in the metric")
+}
+
+func setIntTelemetryProperty(
+	telemetryProperty *int,
+	candidateFields []string,
+	metric telegraf.Metric,
+	usedFields *[]string) error {
+
+	for _, fieldName := range candidateFields {
+		if metric.HasField(fieldName) {
+			metricValue, err := toInt(metric.Fields()[fieldName])
+			if err != nil {
+				return err
+			}
+
+			*telemetryProperty = metricValue
+
+			if usedFields != nil {
+				*usedFields = append(*usedFields, fieldName)
+			}
+
+			return nil
+		}
+	}
+
+	return fmt.Errorf("No field from the candidate list was found in the metric")
+}
+
+func addRemainingMeasurements(telemetry appinsights.Telemetry, metric telegraf.Metric, usedFields []string) {
+	properties := telemetry.GetProperties()
+
+	for fieldName, fieldValue := range metric.Fields() {
+		if contains(usedFields, fieldName) {
+			continue
+		}
+
+		float64Value, err := toFloat64(fieldValue)
+		if err != nil {
+			continue
+		}
+
+		// We would like to add the field to telemetry.Measurements here
+		// But because of issue https://github.com/Microsoft/ApplicationInsights-Go/issues/12
+		// this is currently impossible. So we are just going to add the measurement as a property instead.
+		// This is what we want to be able to do, eventually:
+		//     telemetry.GetMeasurements()[fieldName] = float64Value
+		// (plus handle name conflics)
+		addUniquelyNamedProperty(properties, fieldName, fmt.Sprintf("%g", float64Value))
+	}
+}
+
+func contains(set []string, val string) bool {
+	for _, elem := range set {
+		if elem == val {
+			return true
+		}
+	}
+
+	return false
+}
+
+func addUniquelyNamedProperty(properties map[string]string, name string, value string) {
+	if _, found := properties[name]; !found {
+		properties[name] = value
+	} else {
+		suffix := 1
+		for {
+			_, found = properties[name+"_"+strconv.Itoa(suffix)]
+			if !found {
+				break
+			}
+			suffix++
+		}
+		properties[name+"_"+strconv.Itoa(suffix)] = value
+	}
+}
+
+func toFloat64(value interface{}) (float64, error) {
+	switch v := value.(type) {
+	case int:
```

Comment:
Can use list in the case statement
case int, int8, int16...:

Reply:
See [type switches](https://golang.org/doc/effective_go.html#type_switch)

Reply:
Meaning, you can't